docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,131 @@
+# ChiselEnum
+
+The ChiselEnum type is an easy way to get away from value encodings and reduce the chance of error when encoding muxes, opcodes, and functional unit operations. As opposed to the Chisel type Enums, ChiselEnums allow for names to be explicitly passed through Bundles like IO. 
```

Comment:
```suggestion
The ChiselEnum type can be used to reduce the chance of error when encoding mux selectors, opcodes, and functional unit operations. In contrast with`Chisel.Enum`, `ChiselEnum` can be included in `Bundle`s.
```

I still don't know what you mean by "names to be explicitly passed" or "through Bundles like IO". Even from an english language level, I don't know if "like IO" means "Bundles which are similar to IO" or "similar to the way IO can be passed through Bundles" Is the above correct?

Reply:
Well I wanted to explicitly mention `IO` because it is the most prevalent use of `Bundle`s. When I was first trying to use `Enum` for my mux selects, I couldn't pass the `Enum` type through the `IO` of my modules. This is only possible with `ChiselEnum`s which define a `Data` type that can be passed through Bundles and IO. Since `IO` takes in a `Bundle` object, I was trying to make it clear that `ChiselEnum`s can be passed through `Bundle`s like those used in `IO` while `Enum` objects cannot

Reply:
thanks for the explanation! What do you think of my suggested change above? (Feel free to use any/all/none of it)

Reply:
It looks good. The only thing that I would change is that `Enum` is in the `Chisel.util` package, not in `Chisel`. I don't know if you were trying to reference the package or not.

Reply:
yeah I was trying to reference the package to be explicit what we meant by "Enum". I'll fix it to say util